packages/astro/components/ViewTransitions.astro

```diff
@@ -187,6 +174,22 @@ const { fallback = 'animate' } = Astro.props as Props;
 				html.setAttribute(name, value)
 			);
 
+			// Replace scripts in both the head and body.
+			for(const s1 of document.scripts) {
+				for (const s2 of doc.scripts) {
```

Comment:
Any chance to have a better name? `document` and `doc` are the same to me 😅

Reply:
Yeah, it's really the new document, i'll rename

Reply:
Well actually, it might be better to do that in https://github.com/withastro/astro/pull/8571 since that's a refactor. cc @martrapp

Reply:
Please feel free to make those chances directly on the current source. I will also do the improvements that ematipico suggested for # 8571 on the current state and update the refactoring just before 3.2

Reply:
Thank you